src/Simplex/Chat.hs

```diff
@@ -3510,20 +3510,37 @@ processAgentMessageConn vr user@User {userId} corrId agentConnId agentMessage =
       CON _ -> Just ConnReady
       _ -> Nothing
 
+    processCONFpqSupport :: Connection -> PQSupport -> m Connection
+    processCONFpqSupport conn@Connection {connId, pqSupport} pqSupport' =
+      case (pqSupport, pqSupport') of
+        (PQSupportOn, PQSupportOff) -> do
+          withStore' $ \db -> updateConnSupportPQ db connId pqSupport'
+          pure (conn {pqSupport = pqSupport', pqEncryption = CR.pqSupportToEnc pqSupport'} :: Connection)
+        (PQSupportOff, PQSupportOn) -> do
+          messageWarning "processCONFpqSupport: unexpected PQSupportOn"
+          pure conn
+        _ -> pure conn
```

Comment:
```suggestion
    processCONFpqSupport conn@Connection {connId, pqSupport = pq} pq' =
      | pq == PQSupportOn && pq' == PQSupportOff) = do
          withStore' $ \db -> updateConnSupportPQ db connId pq'
          pure (conn {pqSupport = pq', pqEncryption = CR.pqSupportToEnc pqSupport'} :: Connection)
      | pq /= pq' =
          conn <$ messageWarning "processCONFpqSupport: unexpected PQSupportOn"
      | otherwise = pure conn
```